File: sorting-algorithms/bubbleSort.js
function swap(array, index1, index2) {
  var temp = array[index1];
  array[index1] = array[index2];
  array[index2] = temp;
}

function bubbleSort(array){
    for(let i = 0, arrayLength = array.length; i<arrayLength; i++){
        for(let j = i; j<=arrayLength; j++){
            if(array[i]>array[j]){
                swap(array,i, j);
            }
        }
    }
    return array;
}

console.log(bubbleSort([2,5, 6, 1, 3, 7, 9, 4]));

File: sorting-algorithms/bubble_sort.cpp
// C++ implementation of Bubble sort 
#include <bits/stdc++.h> 
using namespace std; 

void swap(int *xp, int *yp) 
{ 
	int temp = *xp; 
	*xp = *yp; 
	*yp = temp; 
} 

// Bubble Sort function 
void bubbleSort(int arr[], int n) 
{ 
	int i, j; 
	for (i = 0; i < n-1; i++)	 
	
	for (j = 0; j < n-i-1; j++) 
		if (arr[j] > arr[j+1]) 
			swap(&arr[j], &arr[j+1]); 
} 

// Print array function
void printArray(int arr[], int size) 
{ 
	int i; 
	for (i = 0; i < size; i++) 
		cout << arr[i] << " "; 
	cout << endl; 
} 

int main() 
{ 
	int arr[] = {58, 28, 12, 34, 92, 1, 90}; 
	int n = sizeof(arr)/sizeof(arr[0]); 
	bubbleSort(arr, n); 
	cout<<"Sorted array: \n"; 
	printArray(arr, n); 
	return 0; 
} 
